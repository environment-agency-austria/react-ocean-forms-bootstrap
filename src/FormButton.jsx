import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'reactstrap';
import { withForm } from 'react-ocean-forms';

/**
 * Wrapper for a bootstrap button that will
 * automatically disable the button if the
 * form is busy
 */
function FormButton(props) {
  const {
    context: {
      busy,
      disabled,
    },
    ...rest
  } = props;

  const buttonDisabled = busy || disabled;

  return (
    <Button disabled={buttonDisabled} {...rest} />
  );
}

FormButton.propTypes = {
  context: PropTypes.shape({
    busy: PropTypes.bool.isRequired,
    disabled: PropTypes.bool.isRequired,
  }).isRequired,
};

export const BaseFormButton = FormButton;
export default withForm(FormButton);
