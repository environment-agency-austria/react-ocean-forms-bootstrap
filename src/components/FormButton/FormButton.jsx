/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { Button } from 'reactstrap';
import { FormButton as BaseFormButton } from 'react-ocean-forms';

/**
 * Wrapper for a bootstrap button that will
 * automatically disable the button if the
 * form is busy
 */
function FormButton(props) {
  return (
    <BaseFormButton component={Button} {...props} />
  );
}

FormButton.displayName = 'FormButton';

export default FormButton;
