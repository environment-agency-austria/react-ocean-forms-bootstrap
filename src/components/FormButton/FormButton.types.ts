import {FormButton, PropsOf, Subtract } from 'react-ocean-forms';
import { ButtonProps } from 'reactstrap';

type TUpstreamButtonProps = ButtonProps & JSX.LibraryManagedAttributes<typeof FormButton, PropsOf<typeof FormButton>>;

/**
 * Props for the FormButton component
 */
export interface IFormButtonProps extends Subtract<TUpstreamButtonProps, { component: string }> { }
