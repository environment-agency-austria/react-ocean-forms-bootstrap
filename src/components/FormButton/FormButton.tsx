/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as React from 'react';

import { FormButton as BaseFormButton } from 'react-ocean-forms';
import { Button } from 'reactstrap';

import { IFormButtonProps } from './FormButton.types';

/**
 * Wrapper for a bootstrap button that will
 * automatically disable the button if the
 * form is busy
 */
export class FormButton extends React.Component<IFormButtonProps> {
  public static displayName: string = 'FormButton';

  public render(): JSX.Element {
    return (
      <BaseFormButton component={Button} {...this.props} />
    );
  }
}
