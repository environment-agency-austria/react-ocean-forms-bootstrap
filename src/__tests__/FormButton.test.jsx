import React from 'react';
import { shallow } from 'enzyme';

import { BaseFormButton } from '../FormButton';

describe('<FormButton />', () => {
  const wrapper = shallow((
    <BaseFormButton
      context={{ busy: false, disabled: false }}
    />
  ));

  it('should render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should be disabled when the form is busy', () => {
    wrapper.setProps({
      context: {
        busy: true,
        disabled: false,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('should be disabled when the form is disabled', () => {
    wrapper.setProps({
      context: {
        busy: false,
        disabled: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
