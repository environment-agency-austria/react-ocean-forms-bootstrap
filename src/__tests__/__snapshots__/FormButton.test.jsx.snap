// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormButton /> should be disabled when the form is busy 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormButton
    context={
      Object {
        "busy": true,
        "disabled": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "color": "secondary",
      "disabled": true,
      "tag": "button",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "secondary",
        "disabled": true,
        "tag": "button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<FormButton /> should be disabled when the form is disabled 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormButton
    context={
      Object {
        "busy": false,
        "disabled": true,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "color": "secondary",
      "disabled": true,
      "tag": "button",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "secondary",
        "disabled": true,
        "tag": "button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<FormButton /> should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormButton
    context={
      Object {
        "busy": false,
        "disabled": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "color": "secondary",
      "disabled": false,
      "tag": "button",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "color": "secondary",
        "disabled": false,
        "tag": "button",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
